perf/go_trace.go

```diff
@@ -0,0 +1,82 @@
+package perf
+
+import (
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"os"
+	"runtime/trace"
+	"strconv"
+	"sync"
+	"time"
+)
+
+var traceMetricsEnabled, _ = getEnvBool("METRICS_GO_TRACE_ENABLED")
+
+var mutex sync.Mutex
+var running bool
+var file *os.File
+
+func StartGoTraceMetrics() {
+	if !traceMetricsEnabled {
+		return
+	}
+	startTrace()
+	go func() {
+		router := gin.Default()
+		router.GET("/trace/start", func(c *gin.Context) {
+			startTrace()
```

Comment:
Shall we pass a trace duration? After that the node will stop trace automatically, so we dont have to call the `stop` again,

Reply:
added duration